### PR TITLE
Dialyzer fixes for oc_chef_authz_acl

### PR DESCRIPTION
**Pull request by joedevivo**

### CODE REVIEW

File: dev/README.md [hunk 1]
```diff
@@ -85,6 +85,45 @@ available.
 
 Ruby project dependency loading support coming soon.
 
+### Code Coverage
```
Comment (stevendanna) [inline dev/README.md]: This is great, thanks for adding it.

**Comment by marcparadise** [comment]
:+1:

**Comment by stevendanna** [comment]
:+1:

File: dev/README.md [hunk 2]
```diff
@@ -85,6 +85,45 @@ available.
...
+
+or, for all the modules:
+
+```erlang
+Dirs = file:wildcard("/host/oc_erchef/apps/*/ebin") .
+[cover:compile_beam_dir(Dir) || Dir <- Dirs].
+```
+
+#### Run Pedant
+or whatever
```
Comment (tylercloke) [inline dev/README.md]: This feels a bit weird to have in a readme. Not quite sure what is being communicated here either.

**Comment by tylercloke** [comment]
Looks good other than my one comment.

File: dev/README.md (hunk 2 above)
Comment (joedevivo) [inline dev/README.md]: Every time I try to run coverage, I have to go scroll back in a conversation I had with @marcparadise in hipchat. at least this helps me find the commands until a point in the future when Marc is going to (hopefully) give us a script for this

Comment (tylercloke) [inline dev/README.md]: I'm just unsure what this addition means or does. Maybe add a short paragraph after `### Code Coverage`?

**Comment by stevendanna** [comment]
@jd Seems you are still working on this branch.  Is the plan to hold off on merging it until the dialyzer work is done?  Wondering if I should submit a PR for the small dvm fix separately.

**Comment by jd** [comment]
@stevendanna You meant @joedevivo I guess :)

**Comment by manderson26** [comment]
:+1: This makes me very happy to see.

File: src/oc_erchef/custom.mk [hunk 3]
```diff
@@ -14,10 +14,6 @@ DIALYZER_OPTS =
 DIALYZER_SRC = -r apps/chef_db/ebin -r apps/chef_index/ebin -r apps/chef_objects/ebin -r apps/depsolver/ebin -r apps/oc_chef_authz/ebin -r apps/oc_chef_wm/ebin
 DIALYZER_SKIP_DEPS = couchbeam
 
-## TODO WE REALLY SHOULDN'T SKIP THIS
-## ETOOMANYERRORS to fix right now
-SKIP_DIALYZER ?= true
-
```
Comment (stevendanna) [inline src/oc_erchef/custom.mk]: :sparkles: :star2:

File: dev/README.md [hunk 4]
```diff
@@ -85,6 +85,37 @@ available.
...
+```
+
+#### You're now "covered"
+Test chef server however you'd like. A good way would be running pedant as described above, but if you want to test a specific scenario, now's your chance.
+
+#### Generate a report
+Once you're done running whatever tests you want to do, you'll need to generate coverage reports.
+
+For each module you want to generate a report for `cover:analyze_to_file(Module, "/root/filename").`
+```
```
Comment (sdelano) [inline dev/README.md]: This looks like an extra set of code markers and shows up in the rendered diff as such.

**Comment by joedevivo** [comment]
OMG as an added bonus, `warnings_as_errors` is back on!

**Comment by tylercloke** [comment]
Hells to the yes dude. :100:

**Comment by metadave** [comment]
:doughnut: :hamburger:

**Pull request merged**